### DISCUSSION TITLE
Region_size Option in the Pathology Example

### DISCUSSION THREAD

**Discussion by tsaris** [Project-MONAI/MONAI#3790] [Q&A] [answered]
First thank you for having MONAI publicly, it is great!
I am looking the tutorial pathology example that you have, and I am wondering how much I can rely on the region size option? https://github.com/Project-MONAI/tutorials/blob/master/pathology/tumor_detection/torch/camelyon_train_evaluate_pytorch_gpu.py#L458 
The used coordinates seems to be taken from here: https://github.com/baidu-research/NCRF/tree/master/coords that as far as I can tell they are only for 768x768. 

If for example I use MONAI's pathology example with `--region_size`  for 1024x1024 on the CAMELYON16 dataset the patches won't be correct because the coordinate file mentioned in the README.md (https://drive.google.com/file/d/1m2pwko6hxwsxeDWZY2oSOV-_KT97Ol0o/view?usp=sharing) would correspond to 768x768. Am I reading this right? 
Thank you so much!

**Comment by Nic-Ma** [comment]
Hi @ZiyueXu77 ,

Could you please help share some introduction here?

Thanks in advance.

**Comment by ZiyueXu77** [comment, marked as answer]
Hi @tsaris,
Yes, your understanding is correct: the example coordinate file "dataset_0.json" only works with 3x3 grid of 768x768. In order to use a different region/grid_size, we need to generate a new json file to "pre-fetch" the class information from the WSI annotations.
Thanks for the question! This could be a future improvement.

**Reply by tsaris** [reply]
Thank you so much for your quick response, if there could be an "automatic" option to generate larger patches it would be great. Also if you have a code on how to generate the json file for arbitrary number of patches, let's say 1024x1024 and you can share it would be great too!

**Reply by ZiyueXu77** [reply]
Agreed! @drbeh we may include the json generation script and instructions for the completeness of this pipeline.

**Reply by bhashemian** [reply]
Hi @tsaris, thanks for bringing up this discussion. This process has two steps for a given patch size, first it needs to derive the patch-wise annotations from polygon annotations, and then convert then into the json structure that we are using. Although we would definitely want to provide the whole functionality, for this tutorial we are currently relying on the coordinations from NCRF repo. You can refer to their repo on how to derive the coordinations for the patch size of your choice, and once you have them, I can help you to convert them to the desired json structure.

**Reply by tsaris** [reply]
Hi @drbeh the NCRF repo doesn't provide a way to generate the coordinates either: https://github.com/baidu-research/NCRF/issues/29 So with their code too, I don't think you can go higher than  768x768 patch size, or probably use something other than  CAMELYON16 dataset.

**Reply by bhashemian** [reply]
I see! I was expecting that it has been addressed after few years. Anyways, I have created a feature request for that in MONAI https://github.com/Project-MONAI/MONAI/issues/3848 but it goes to the queue of the tasks for digital pathology and will be assigned a priority level by the pathology working group. Stay tuned!

**Reply by tsaris** [reply]
Thanks! Please feel free to contact me (tsarisa@ornl.gov) if there is help need it on this. I am interested on this functionality and using MONAI.
